tests/test_writers.py

```diff
@@ -138,3 +138,7 @@ def test_store_dataset_fragment(temp_store):
     ds_target = xr.open_dataset(temp_store, engine="zarr").load()
 
     xr.testing.assert_identical(ds, ds_target)
+    # assert_identical() doesn't check encoding
+    # Checking the original time encoding units should be sufficient
+    # Zarr retains the original "days since %Y:%m%d" and removes " %H:%M:%S"
+    assert " ".join(ds.time.encoding["units"].split(" ")[0:-1]) == ds_target.time.encoding["units"]
```

Comment:
Zarr has nothing to do with it. Xarray and cftime are managing this logic. I think it drops of the time if the time is 0:0:0.

Wouldn't it be easier to fix this at the source (i.e. strip the time in line 39 of `data_generation.py`)? That way you can just do

```python
assert ds.time.encoding == ds_target.time.encoding
```

Reply:
I originally stripped it in `data_generation.py` as you suggest, but I was sure that this was causing failures only in `test_writer.py`. I've restored this and all tests are passing, so I'm not sure what went wrong previously. Commit is on the way.